docs/proposals/docs-process.md

```diff
@@ -0,0 +1,94 @@
+# Docs Process
+
+@johndmulhausen
+
+A new design for the kubernetes.io site is currently being implemented and we think it's a good time to streamline and simplify the docs creation process as we rebuild the site from scratch.
+
+When a community is empowered to be effective contributors, great things happen. That’s why I’m proposing that we move the documentation from the docs/ folder in the core Kubernetes repo in GitHub into the GitHub pages repo itself, migrate that repo into one named `kubernetes.github.io` so that it will be compatible with the GitHub Pages auto-update system, and leave all munging scripts behind.
+
+**TL;DR: Right now we do a lot of scripting and branching to create docs -- this proposal is about using Jekyll's native capability and GitHub Page's staging capability to create a much easier-to-edit set of documentation that will ship with our upcoming new site.**
```

Comment:
Sorry, but this terrifies me. Many of our docs are autogenerated
completely, and people who do things that change the docs *do not* in
general also view those docs other than via the markdown diff. The culture
change you're asking for here is very large.

On Tue, Feb 9, 2016 at 3:52 PM, johndmulhausen <notifications@github.com>
wrote:

> In docs/proposals/docs-process.md
> <https://github.com/kubernetes/kubernetes/pull/20874#discussion_r52395221>
> :
>
> > @@ -0,0 +1,94 @@
> > +# Docs Process
> > +
> > +@johndmulhausen
> > +
> > +A new design for the kubernetes.io site is currently being implemented and we think it's a good time to streamline and simplify the docs creation process as we rebuild the site from scratch.
> > +
> > +When a community is empowered to be effective contributors, great things happen. That’s why I’m proposing that we move the documentation from the docs/ folder in the core Kubernetes repo in GitHub into the GitHub pages repo itself, migrate that repo into one named `kubernetes.github.io` so that it will be compatible with the GitHub Pages auto-update system, and leave all munging scripts behind.
> > +
> > +**TL;DR: Right now we do a lot of scripting and branching to create docs -- this proposal is about using Jekyll's native capability and GitHub Page's staging capability to create a much easier-to-edit set of documentation that will ship with our upcoming new site.**
>
> Mungers have to go. If we had a system that allowed easy staging of the
> site -- like the one I'm proposing -- we could catch issues with Markdown
> formatting on our own through simple trial and error like anyone else who
> writes Markdown. A system that is this script dependent is too closed and
> fighty to even get started on. I'd prefer to relearn these "lessons" and
> implement solutions that used native capability rather than throwing a
> barely-if-at-all documented script at every formatting problem we see. I'd
> wager a good number of these problems can probably be solved by simply
> writing and circulating some style/editorial guides with good examples.
>
> I'm starting with zero munging scripts in the new repo. Just saying, be
> prepared.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/20874/files#r52395221>.
>